server/src/updaters/relationship-updaters.js

```diff
@@ -91,14 +96,64 @@ async function updateRelationships(
       dbQuery(undirectedInsertQuery),
       dbQuery(directedInsertQuery),
     ];
-
     if (directedDeleteQuery.length) {
       promises.push(dbQuery(directedDeleteQuery));
     }
 
     await Promise.all(promises);
 
     return { ...errors, ...friendRequestErrors };
+  } else if (action === relationshipActions.UNFRIEND) {
+    const updateRows = userIDs.map(userID => {
+      const ids = [userID, viewer.userID].map(Number).sort((a, b) => a - b);
+      return [...ids, undirectedStatus.KNOW_OF];
+    });
+    const insertQuery = SQL`
+      INSERT INTO relationships_undirected (user1, user2, status)
+      VALUES ${updateRows}
+      ON DUPLICATE KEY UPDATE status = VALUES(status)
+    `;
+    const deleteQuery = SQL`
+      DELETE FROM relationships_directed
+      WHERE status = ${directedStatus.PENDING_FRIEND} AND 
+        user1 = ${viewer.userID} AND user2 IN (${userIDs})
+    `;
+
+    await Promise.all([dbQuery(insertQuery), dbQuery(deleteQuery)]);
+
+    return errors;
+  } else if (action === relationshipActions.BLOCK) {
+    const directedRows = [];
+    const undirectedRows = [];
+    for (const userID of userIDs) {
+      const ids = [userID, viewer.userID].map(Number).sort((a, b) => a - b);
```

Comment:
Nit: maybe move `const ids` declaration down one line, since it's only used for `undirectedRows`